src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,59 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+    private static final int FIELD_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private static final String COMMA = ",";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFile(fromFileName);
+        String report = createReport(dataFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    public String readFile(String fromFileName) {
+        File fromFile = new File(fromFileName);
+        try {
+            return Files.readString(fromFile.toPath());
```

Comment:
also not critical, but we can inline this
```suggestion
        try {
            return Files.readString(new File(fromFileName).toPath());
```